src/utils/string/index.ts

```diff
@@ -0,0 +1,23 @@
+export const msToDuration = (ms: number): string => {
+  const days = Math.floor(ms / (24 * 60 * 60 * 1000));
+  const daysMilli = ms % (24 * 60 * 60 * 1000);
+  const hours = Math.floor(daysMilli / (60 * 60 * 1000));
+  const hoursMilli = ms % (60 * 60 * 1000);
+  const minutes = Math.floor(hoursMilli / (60 * 1000));
+  const minutesMilli = ms % (60 * 1000);
+  const seconds = Math.floor(minutesMilli / 1000);
+  const milliseconds = ms % 1000;
+  if (days > 1) {
+    return `${Math.trunc(days)}d ${hours}h ${minutes}m ${seconds}s`;
```

Comment:
I think this is excessive - nothing should really take more than a day, and if it does it'll be a small number of days. I think display a # of hours > 24 in that scenario would be easier to comprehend